Bottom: b640fcf4f0f69f8cb52a36ad028ac0109aeae658
Top:    5177e529dfd053f70023a5a16cd2caa7f86ce70d
Author: Robert Berger <robert.berger@ReliableEmbeddedSystems.com>
Date:   2020-11-13 15:40:43 +0200

ipk: package/package feed GPG signing

1) do_package_index[depends] += "signing-keys:do_deploy"

2) do_rootfs[depends] += "signing-keys:do_populate_sysroot"

3) use gnupg-native instead of the host version
   the host version (e.g. the one with Ubuntu 18)
   runs out of "secure memory" when not invoked sequentially.
   And we all know how BitBake likes to invoke stuff in parallel;)
   see e.g https://dev.gnupg.org/T2857

Signed-off-by: Robert Berger <robert.berger@ReliableEmbeddedSystems.com>


---

diff --git a/meta/classes/sign_ipk.bbclass b/meta/classes/sign_ipk.bbclass
index e5057b7799..15f63257b7 100644
--- a/meta/classes/sign_ipk.bbclass
+++ b/meta/classes/sign_ipk.bbclass
@@ -50,3 +50,9 @@ def sign_ipk(d, ipk_to_sign):
                        d.getVar('IPK_GPG_NAME'),
                        d.getVar('IPK_GPG_PASSPHRASE_FILE'),
                        armor=is_ascii_sig)
+
+
+do_package_index[depends] += "signing-keys:do_deploy"
+do_rootfs[depends] += "signing-keys:do_populate_sysroot"
+
+PACKAGE_WRITE_DEPS += "gnupg-native"
